internal/utils.go

```diff
@@ -0,0 +1,72 @@
+package internal
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/codecrafters-io/tester-utils/logger"
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+type GitTempDir struct {
+	TempDir     string
+	OriginalDir string
+	TempGitPath string
+	logger      *logger.Logger
+}
+
+// MoveGitToTemp moves the system git binary to a temporary directory
+func MoveGitToTemp(harness *test_case_harness.TestCaseHarness, logger *logger.Logger) (*GitTempDir, error) {
```

Comment:
```suggestion
func MoveGitToTemp(harness *test_case_harness.TestCaseHarness, logger *logger.Logger) {
```

- We aren't using the first return value anywhere. 
- All errors are internal errors so we can just panic instead, no need to return an error